sdk/identity/azure-identity/azure/identity/_credentials/default.py

```diff
@@ -256,3 +276,18 @@ def get_token_info(self, *scopes: str, options: Optional[TokenRequestOptions] =
         token_info = cast(SupportsTokenInfo, super()).get_token_info(*scopes, options=options)
         within_dac.set(False)
         return token_info
+
+
+def parse_azure_dac(az_dac, avail_credentials):
+    striped_az_dac = az_dac.strip()
+    if striped_az_dac.endswith(";"):
+        striped_az_dac = striped_az_dac[:-1]
+    creds = [cred.strip() for cred in striped_az_dac.split(";")]
+    credentials = []
+
+    for cred in creds:
+        if cred not in VALID_CREDENTIALS:
+            raise ValueError(f"Invalid credential '{cred}' in AZURE_DEFAULT_CREDENTIAL_ALLOW_LIST")
+        credentials.append(avail_credentials.get(cred))
```

Comment:
Presumably, there can be cases where a credential is valid but not available, so let's only append if not None.
```suggestion
        credential = avail_credentials.get(cred)
        if credential:
            credentials.append(credential)
```

Reply:
We already have the protection in https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/identity/azure-identity/azure/identity/_credentials/chained.py#L110.